src/events/loop.cc

```diff
@@ -410,7 +410,10 @@ void loop::_dispatching() {
                                 nullptr);
       }
 
-      command_manager::instance().execute();
+      auto now = std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::system_clock::now().time_since_epoch()).count(); 
+      long later = now + (config->sleep_time() * 1000000000);
+
+      command_manager::instance().execute(config->sleep_time() * 1000000000);
```

Comment:
Since config->sleep_time is floating point number, you could keep as is. We then know the parameter in execute() is in seconds. No need to make this multiplication. And in the function, you just have to tell the duration is in seconds.